onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -1848,12 +1880,15 @@
         ORT_ENFORCE(execution_plan[node_stream_map_[node_index]]->device_.Type() == node_device_mem_location.device.Type());
       }
     }
-    // 4. set notification owners
+    // 4. set notification owners. Notice that both diverge_stream_from_node and node_to_notification will contribute to notification_owners
     plan_.notification_owners.resize(num_notifications);
     for (auto node_index : graph_viewer_.GetNodesInTopologicalOrder(context_->GetExecutionOrder())) {
-      auto it = node_to_notification.find(node_index);
+      auto it = diverge_stream_from_node.find(node_index);
+      if (it != diverge_stream_from_node.end()) plan_.notification_owners[it->second] = node_stream_map_[node_index];
+
+      it = node_to_notification.find(node_index);
       if (it != node_to_notification.end()) {
-        // notification owned by the node who produced it.
+        // notification owned by the stream who produced it.
         plan_.notification_owners[it->second] = node_stream_map_[node_index];
       }
```

Comment:
Assume "it" are both valid on 1887 and 1890, will it->second will be different or same?

Reply:
It's not possible. diverge_stream_from_node records the case that nodes and their consumers are in different streams, while node_to_notification records the case that nodes and their consumers are in the same stream.

Reply:
I think it deserves a UT for this scenario: a node A is in different stream with one of its consumer node B (so a pair of ActivateNotification/wait steps will be needed), and in the same stream with another of its consumer node C which still needs to launch ActivateNotification. The previous code will miss one pair, I think this PR will handle this case well.